batchflow/config.py

```diff
@@ -1,250 +1,257 @@
+                        raise KeyError(k)
+                if pop:
+                    del parent[k]
+
+            # use dict's 'get' or 'pop' if '/' not in variable
             else:
-                if has_default:
-                    val = default
+                if variable in self:
+                    value = method(variable)
                 else:
-                    raise KeyError(f"Key '{variable}' not found")
+                    if has_default:
+                        return default
+                    raise KeyError(variable)
 
-            val = Config(val) if isinstance(val, (dict, Config.IAddDict)) else val
-            ret_vars.append(val)
+            ret_vars.append(value)
 
-        if unpack:
-            ret_vars = ret_vars[0]
-        else:
-            ret_vars = tuple(ret_vars)
-        return ret_vars
+        ret_vars = ret_vars[0] if unpack else tuple(ret_vars)
 
-    def put(self, variable, value, config=None):
-        """ Put a new variable into config
+        return ret_vars
 
-        Parameters
-        ----------
-        variable : str
-            variable to add. '/' is used to put value into nested dict
-        value : masc
-        config : dict, Config or None
-            if None value will be putted into self.config else from config
-        """
-        if config is None:
-            config = self.config
-        elif isinstance(config, Config):
-            config = config.config
-        if isinstance(value, dict):
-            value = Config(value)
-        variable = variable.strip('/')
-        if '/' in variable:
-            var = variable.split('/')
-            prefix = var[:-1]
-            var_name = var[-1]
-        else:
-            prefix = []
-            var_name = variable
-
-        for i, p in enumerate(prefix):
-            if p not in config:
-                config[p] = Config.IAddDict()
-            if isinstance(config[p], dict):
-                config = config[p]
-            else: # for example, we put value with key 'a/b' into `{a: c}`
-                value = Config({'/'.join(prefix[i+1:] + [var_name]): value})
-                var_name = p
-                break
-        if var_name in config and isinstance(config[var_name], dict) and isinstance(value, Config):
-            config[var_name] = Config(config[var_name])
-            config[var_name].update(value)
-            config[var_name] = config[var_name].config
-        else:
-            if isinstance(value, Config):
-                config[var_name] = value.config
-            else:
-                config[var_name] = value
+    def __delitem__(self, key):
+        self.pop(key)
 
-    def parse(self, config):
-        """ Parses flatten config with slashes
+    def __getattr__(self, key):
+        if key in self:
+            value = self.get(key)
+            value = Config(value) if isinstance(value, dict) else value
+            return value
+        raise AttributeError(key)
+    
```

Comment:
Maybe `AttributeError` should be defined in the `self.get`?

Reply:
But in this case we will catch `AttributeError` every time we want to retrieve the key from the dict, not only by `.`

Reply:
> Maybe `AttributeError` should be defined in the `self.get`?

But in this case we will catch `AttributeError` every time we want to retrieve the key from the dict, not only by `.`